doc/custom_nonconformity_score_proof.ipynb

```diff
@@ -0,0 +1,407 @@
+    "& \\forall i \\neq j: R_{i,j} = f(Y_{i}, \\hat{\\mu}_{-(i,j)}(X_{i}))\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Prediction function\n",
+    "From a conformity score and a prediction, the function $g$ ouputs a perturbed prediction:\n",
+    "$$\\begin{align}\n",
+    "g:F \\times E &\\to E\\\\\n",
+    "(R, \\hat{Y}) &\\mapsto g(R, \\hat{Y})\n",
+    "\\end{align}$$\n",
+    "\n",
+    "It verifies the following properties:\n",
+    "$$\\begin{align}\n",
+    "& \\forall R_{l} < R_{h}: g(R_{l}, \\hat{Y}) < g(R_{h}, \\hat{Y}) \\\\\n",
+    "& \\forall i \\neq j: Y_{i} = g(R_{i,j}, \\hat{\\mu}_{-(i,j)}(X_{i}))\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Function relationship\n",
+    "The two functions must verify the following property:\n",
+    "$$\\begin{align}\n",
+    "\\forall \\hat{Y}, \\forall R: R = f(g(R, \\hat{Y}), \\hat{Y}) \n",
+    "\\end{align}$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88da4fd0",
+   "metadata": {},
+   "source": [
+    "___\n",
+    "# The jackknife+ prediction interval\n",
+    "Let derive the proof of the theorem to build [the jackknife+ prediction interval](https://arxiv.org/pdf/1905.02928.pdf) (Theorem 1, part 6, p. 18-21) with more general nonconformity scores.\n",
+    "\n",
+    "## R contruction\n",
+    "Instead of defining the R matrix with the absolute residual, let assume that R is defined by means of a nonconformity score $f$:\n",
+    "$$\\begin{align}\n",
+    "R_{i,j} &= f(Y_{i}, \\hat{\\mu}_{-(i,j)}(X_{i})) \\\\\n",
+    "R_{i}^{LOO} &= R_{i,n+1}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## A contruction\n",
+    "There are two variants:\n",
+    "- the symmetrical case when the conformity scores are assumed to be symmetrical around 0. In this case, the absolute values of the conformity scores are used to construct the A matrix.\n",
+    "- otherwise, the signed conformity scores are kept as they are.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, the A matrix is defined as below:\n",
+    "$$A_{i,j} = \\mathbb{1} \\{ \\lvert R_{i,j} \\rvert > \\lvert R_{j,i} \\rvert \\}$$\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "The two following definitions of matrix A are considered: \n",
+    "$$\\begin{align}\n",
+    "A^{+}_{i,j} &= \\mathbb{1} \\{R_{i,j} > R_{j,i}\\} \\\\\n",
+    "A^{-}_{i,j} &= \\mathbb{1} \\{R_{i,j} < R_{j,i}\\} = J - A^{+}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Strange points\n",
+    "Like previously, the definition of strange points depends on the case considered.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, there is a single set $S(A)=\\{i \\in \\{1,...,n+1\\}:A_{i.}\\geq(1-\\alpha)(n+1)\\}$.\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "Instead of the single $S(A)$ set, two cases should be considered: $A^{+}$ and $A^{-}$. Keeping the same definition of the set $S$ as the paper, we have:\n",
```

Comment:
the single $S(A)$ set -> the single  set $S(A)$
two cases -> two sets
definition of the set $S$ as the paper -> definition of the set $S$ of the paper

Reply:
- Done
- Done
- Done

Reply:
- Done
- Done
- Done